src/AppProvider.tsx

```diff
@@ -0,0 +1,62 @@
+/**
+ * A wrapper that provides app state and dispatches events
+ */
+
+import React from "react";
+import { useState } from "react";
+import Tinycon from "tinycon";
+
+type AppState = {
+  harvestable: number,
+};
+
+interface AppContext {
+  state: AppState;
+  appDispatcher: (action: AppAction) => AppState;
+}
+
+export const AppContext = React.createContext<AppContext>({} as AppContext);
+
+type AppAction =
+  | {
+      type: "harvestable.increment";
+      value: number;
+    };
+
+function eventReducer(state: AppState, action: AppAction) {
+  if (action.type === "harvestable.increment") {
+
+    if (action.value != -1 && action.value !== 1) {
+      throw new Error("This counter should increment or decrement by 1 only");
+    }
+    const harvestable = state.harvestable + action.value;
+    Tinycon.setBubble(harvestable);
+    return {
+      ...state,
+      harvestable
+    } as AppState;
+  }
+
+  throw new Error(`Unexpected event dispatched`);
+}
+
+export const AppProvider: React.FC = ({ children }) => {
+  const [state, setState] = useState<AppState>({
+    harvestable: 0,
+  });
+
+  const appDispatcher = React.useCallback(
```

Comment:
Given that there is only 1 event, consider going a simpler approach and not using an event reducer.

E.g.

```
(value: number) => {
	if (value != -1 && value !== 1) {
      throw new Error("This counter should increment or decrement by 1 only");
    }
    const harvestable = state.harvestable + value;
    Tinycon.setBubble(harvestable);
    setState({
      ...state,
      harvestable
    } as AppState);
}
```

Reply:
Done, removed useCallback, I think that is not useful because there is a side-effect